pkg/karmadactl/cmdinit/cert/cert_test.go

```diff
@@ -101,7 +103,10 @@ func TestGenCerts(_ *testing.T) {
 	apiserverCertCfg := NewCertConfig("karmada-apiserver", []string{""}, karmadaAltNames, &notAfter)
 	frontProxyClientCertCfg := NewCertConfig("front-proxy-client", []string{}, certutil.AltNames{}, &notAfter)
 
-	if err := GenCerts(TestCertsTmp, etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
+	if err := GenCerts(TestCertsTmp, "", "", etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
+		fmt.Println(err)
+	}
+	if err := GenCerts(TestCertsTmp, TestCaCertPath, TestCaKeyPath, etcdServerCertConfig, etcdClientCertCfg, karmadaCertCfg, apiserverCertCfg, frontProxyClientCertCfg); err != nil {
 		fmt.Println(err)
 	}
 }
```

Comment:
```suggestion

	certFiles := []string{
		"apiserver.crt", "apiserver.key",
		"ca.crt", "ca.key",
		"etcd-ca.crt", "etcd-ca.key",
		"etcd-client.crt", "etcd-client.key",
		"etcd-server.crt", "etcd-server.key",
		"front-proxy-ca.crt", "front-proxy-ca.key",
		"front-proxy-client.crt", "front-proxy-client.key",
		"karmada.crt", "karmada.key",
	}
	if err := checkCertFiles(TestCertsTmp, certFiles); err != nil {
		t.Fatal(err)
	} else {
		klog.Infof("All certificate files are present.")
	}
}

func checkCertFiles(path string, files []string) error {
	for _, file := range files {
		filePath := fmt.Sprintf("%s/%s", path, file)
		if _, err := os.Stat(filePath); os.IsNotExist(err) {
			return fmt.Errorf("cert file not found: %s", filePath)
		}
	}
	return nil
}
```

Reply:
As for the changes I mentioned above, I hereby give my preliminary implementation method. If there is a better implementation method or opinion, please feel free to let me know @guozheng-shen  @liangyuanpeng 
ps: I am not sure whether I should submit in this pr, so I will provide the code in this way for the time being. If I need to modify it, please kindly reply to me

Reply:
@tiansuo114  please create a new issue for it, this PR is not working for it.

Reply:
link to issue https://github.com/karmada-io/karmada/issues/5171